### PR TITLE
[Doc] Add links to client feature matrix in README.md

### PR DESCRIPTION
**Pull request by momo-jun**
### Motivation

The new [client feature matrix](https://pulsar.apache.org/client-feature-matrix/) page has been live, so this PR makes the required updates on the doc side.

### Modifications

Add a paragraph to notify both users and contributors about the new client feature matrix.

### Documentation

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)

- [ ] `doc-not-needed` 
(Please explain why)

- [x] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,28 +49,28 @@ yarn add pulsar-client
 Then you can run the following simple end-to-end example:
 
 ```javascript
-const Pulsar = require('pulsar-client');
+const Pulsar = require("pulsar-client");
```
Comment (momo-jun) [inline README.md:52]: The quotes were auto-corrected, maybe by a checker plugin. Does it make sense or shall I revert?

Comment (shibd) [reply README.md:52]: It better uses `'`.

Comment (shibd) [reply README.md:52]: You should revert it.

Comment (momo-jun) [reply README.md]: Thanks. Resolved.

**Review by shibd — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
Merging...

Thank you!

**Pull request merged**